core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/helper/DefaultLocalDcHelper.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.loadbalancing.helper;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.internal.core.context.InternalDriverContext;
+import com.datastax.oss.driver.internal.core.metadata.DefaultNode;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultLocalDcHelper extends BasicLocalDcHelper {
```

Comment:
I find the terminology a bit confusing, although I'm not sure I have better names to suggest.
* "inferring": that one is clear
* "basic": does not require the local DC, except if it's provided (why?). If it completely ignored it maybe it could be `NoLocalDcHelper`.
* "default": maybe `MandatoryLocalDcHelper`

Reply:
Ah names :-) `MandatoryLocalDcHelper` works for me. `NoLocalDcHelper` annoys me however because it not always return an empty optional. Would `OptionalLocalDcHelper` work for you?